src/requests/typings/deliveryFunction.ts

```diff
@@ -0,0 +1,65 @@
+// Remove when this eslint rule covers all the cases
+// https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/ROADMAP.md
+/*eslint-disable no-unused-vars*/
+
+export enum DeliveryFunctionRequestEventType {
+  GRAPHQL_FIELD_MAPPING = 'graphql.field.mapping',
+  GRAPHQL_QUERY = 'graphql.query',
+}
+
+export type GraphQLFieldTypeMappingRequest = {
+  fields: { contentTypeId: string; field: Field }[]
+}
+
+type Field = {
+  id: string
+  type: string
+}
+
+export type GraphQLFieldTypeMapping = {
+  contentTypeId: string
+  fieldId: string
+  graphQLOutputType: string
+  graphQLQueryField: string
+  graphQLQueryArgument: string
+}
+
+export type GraphQLFieldTypeMappingResponse = GraphQLFieldTypeMapping[]
+
+export type GraphQLQueryRequest = {
+  query: string
+  variables: Record<string, unknown>
+  operationName?: string
+}
+
+/**
+ * @see https://spec.graphql.org/October2021/#sec-Response
+ */
+export type GraphQLQueryResponse = {
+  data?: Record<string, any>
+  errors?: readonly Record<string, any>[]
+  extensions?: Record<string, unknown>
+}
+
+/**
+ * P: Possibility to type app installation parameters
+ */
+export type DeliveryFunctionEventContext = {
+  spaceId: string
+  environmentId: string
+  appInstallationParameters: Record<string, any>
+  originalRequest: {
+    headers: Record<string, string>
```

Comment:
Should we add this in the first version? I feel like headers were necessary in shopify/commercetools but not sure if it makes sense for all custom resolvers